### PR TITLE
testbuild.sh: Make the use of "git clean" optional

### PR DESCRIPTION
**Pull request by yamt**

### CODE REVIEW

File: tools/README.txt [hunk 1]
```diff
@@ -994,18 +994,24 @@ testbuild.sh
 
     $ ./testbuild.sh -h
 
-    USAGE: ./testbuild.sh [-l|m|c|u|g|n] [-d] [-x] [-j <ncpus>] [-a <appsdir>] [-t <topdir>] <testlist-file>
+    USAGE: ./testbuild.sh [-l|m|c|u|g|n] [-si|-sl>] [-d] [-x] [-j <ncpus>] [-a <appsdir>] [-t <topdir>] <testlist-file>
```
Comment (xiaoxiang781216) [inline tools/README.txt:997]: Revert the change?

File: tools/README.txt [hunk 2]
```diff
@@ -994,18 +994,24 @@ testbuild.sh
...
-      -a <appsdir> provides the relative path to the apps/ directory.  Default ../apps
-      -t <topdir> provides the absolute path to top nuttx/ directory.  Default $PWD/../nuttx
-      -p only print the list of configs without running any builds
-      -j <ncpus> passed on to make.  Default:  No -j make option
+         Ubuntu under Windows 10 (u), MSYS/MSYS2 (g) or Windows native (n).  Default Linux
       -d enables script debug output
       -x exit on build failures
+      -j <ncpus> passed on to make.  Default:  No -j make option.
+      -a <appsdir> provides the relative path to the apps/ directory.  Default ../apps
+      -t <topdir> provides the absolute path to top nuttx/ directory.  Default /Users/yamamoto/git/nuttx/nuttx/tools/../nuttx
```
Comment (xiaoxiang781216) [inline tools/README.txt:1007]: Reword /Users/yamamoto/git/nuttx/nuttx/tools/../nuttx

File: tools/README.txt (hunk 1 above)
Comment (yamt) [reply tools/README.txt:997]: will do

File: tools/README.txt (hunk 2 above)
Comment (yamt) [reply tools/README.txt:1007]: oops

**Comment by xiaoxiang781216** [comment]
LGTM, @hartmannathan @btashton if you don't have any concern, I will merge it after the github pass the build.

**Comment by hartmannathan** [comment]
> LGTM, @hartmannathan @btashton if you don't have any concern, I will merge it after the github pass the build.

I don't have any concerns. LGTM, I like the new command line option.

**Comment by yamt** [comment]
the failure doesn't seem related to the change. what's the proper way to re-run the ci?

> /usr/bin/docker: error pulling image configuration: received unexpected HTTP status: 504 Gateway Time-out.
> See '/usr/bin/docker run --help'.

**Comment by btashton** [comment]
@yamt I just restarted the build looks like it got past whatever that error was.

File: tools/README.txt [hunk 3]
```diff
@@ -994,18 +994,26 @@ testbuild.sh
...
+         Ubuntu under Windows 10 (u), MSYS/MSYS2 (g) or Windows native (n).  Default Linux
       -d enables script debug output
       -x exit on build failures
+      -j <ncpus> passed on to make.  Default:  No -j make option.
+      -a <appsdir> provides the relative path to the apps/ directory.  Default ../apps
+      -t <topdir> provides the absolute path to top nuttx/ directory.
+         Default $WD/../nuttx, where $WD is the parent directory of
+         the directory where this script is.
+      -p only print the list of configs without running any builds
+      -G Use "git clean -xfdq" instead of "make distclean" to clean the tree.
```
Comment (davids5) [inline tools/README.txt:1011]: @xiaoxiang781216 I have been burned by this sort of thing, killing dev tool artifacts. Should this warn a user on a non CI platform?

Comment (xiaoxiang781216) [reply tools/README.txt:1011]: By default we sue make distclean. if user add -G option, she/he should know the side effect. The help also indicate this:
```
    -G Use "git clean -xfdq" instead of "make distclean" to clean the tree.
         This option may speed up the builds. However, note that:
           * This assumes that your trees are git based.
           * This assumes that only nuttx and apps repos need to be cleaned.
           * If the tree has files not managed by git, they will be removed
             as well.
```
